ext/pdo_firebird/firebird_driver.c

```diff
@@ -809,28 +805,85 @@ static bool firebird_handle_begin(pdo_dbh_t *dbh) /* {{{ */
 }
 /* }}} */
 
-/* called by PDO to commit a transaction */
-static bool firebird_handle_commit(pdo_dbh_t *dbh) /* {{{ */
+/* called by PDO to start a transaction */
+static bool firebird_handle_manually_begin(pdo_dbh_t *dbh) /* {{{ */
 {
 	pdo_firebird_db_handle *H = (pdo_firebird_db_handle *)dbh->driver_data;
 
-	if (isc_commit_transaction(H->isc_status, &H->tr)) {
-		firebird_error(dbh);
+	if (dbh->auto_commit && H->tr && !firebird_commit_transaction(dbh, FB_TXN_RELEASE)) {
```

Comment:
Is this ``firebird_commit_transaction()`` function part of the FireBird API or the PHP Driver?

Reply:
No, it's a replacement of `_firebird_begin_transaction` with an h file. It's quite difficult to understand....
As you pointed out, I should think about the naming a little more.